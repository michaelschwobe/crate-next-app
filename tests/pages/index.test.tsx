import { render, screen, fireEvent } from '../../.jest/jest.setup';
import Home from '../../src/pages/index';

// -----------------------------------------------------------------------------

describe('Home', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('clicking button triggers alert', () => {
    render(<Home />);
    window.alert = jest.fn();
    fireEvent.click(screen.getByText('Test Button'));
    expect(window.alert).toHaveBeenCalledWith('With typescript and Jest');
  });
});
