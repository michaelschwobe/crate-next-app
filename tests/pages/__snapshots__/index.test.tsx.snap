// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="description"
      >
        Get started by editing 
        <code
          class="code"
        >
          pages/index.js
        </code>
      </p>
      <div
        class="grid"
      >
        <a
          class="card"
          href="/about"
        >
          <h2>
            About Page →
          </h2>
          <p>
            Cypress will E2E test if this link is working.
          </p>
        </a>
        <div
          class="card"
        >
          <h2>
            Unit Tests & Snapshots
          </h2>
          <p>
            Jest will test if this button is working. 
            <button
              type="button"
            >
              Test Button
            </button>
          </p>
        </div>
      </div>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h2>
            Documentation →
          </h2>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h2>
            Learn →
          </h2>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h2>
            Examples →
          </h2>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h2>
            Deploy →
          </h2>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <span
          class="logo"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzIiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Vercel Logo"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
        </span>
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
